src/phase0/state_transition/mod.rs

```diff
@@ -975,7 +975,7 @@ pub fn get_total_balance<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    indices: HashSet<ValidatorIndex>,
+    indices: &HashSet<ValidatorIndex>,
```

Comment:
this would work....

another option is to "erase" the specific type with the existential typing i mentioned above...

```suggestion
    indices: impl Iterator<Item=ValidatorIndex>,
```

and then below you could do `HashSet::from_iter(...).iter()` (which we do bc we want to ensure set semantics at that call site)

and then up above you could do something like:

```rust
let attesting_balance = get_total_balance(state, unslashed_attesting_indices.iter(), context)?;
```

which is functionally the same thing....

.... after typing it out, it seems fine enough to just keep it how you have it. it follows the spec and i don't see any crazy performance differential -- if we accepted more types of inputs to this function or used it in more places in more ways this could change but this PR introduces the only other use of `get_total_balance`